### DISCUSSION TITLE
pre and post media channel contributions after optimization

### DISCUSSION THREAD

**Discussion by xijianlim** [google/lightweight_mmm#32] [Ideas]
Hi,

It would be a great idea to show the media contributions to the KPI for each respective channel as a "before and after" optimization.

If this cannot be developed, would you suggest looping through the "_objective_function" function with zeros and each channel arrays?

**Comment by pabloduque0** [comment]
@xijianlim the example notebooks already show that (we added it a few weeks ago :) )

Let me know if that is what you are looking for.

**Reply by xijianlim** [reply]
@pabloduque0: Your examples only show the allocation of budgets per channel, but not each channels' contribution to the target.

As you can see: there's only spend and an aggregated contribution.

It would be really useful to see a similar plot of the budget allocation plot above, but the y-axis shows contribution per channel on a pre/post optim basis.


![image](https://user-images.githubusercontent.com/25444256/177227255-58bd7396-2de5-40d7-a3aa-3b4f5eb0da46.png)

**Reply by pabloduque0** [reply]
The media contribution is generally calculated on past data, you can technically calculate it on the future predictions but it might not make too much sense to do so.

The effect of media on KPI wont change after optimisation, the only thing that changes is the budget allocation. What was learned by the model will keep applying to any optimisation or scenario you run.

What is that you are trying to achieve exactly?

**Reply by xijianlim** [reply]
I'm trying to breakdown the pre and post optimization by channel in terms of each media's contribution to the KPI. This is very typical request in the industry for MMMs.  This is an example of what I'm trying to achieve: 
![image](https://user-images.githubusercontent.com/25444256/177398305-42dd7c51-c372-4b9e-89b9-36faa801fb3e.png)

This could be done by splitting up the pre and post optimization predicted targets by each media channel's contribution to target.

**Reply by xijianlim** [reply]
Also as an aside, a very typical request is to see scenarios where the total budget varies (-10% /+10%).

**Reply by pabloduque0** [reply]
Oh I see what you mean.

Yes that is definitely doable. We might be able to do a general case for obtaining media contribution for any prediction, but might not be something coming in a super immediate future. But we are adding it to our list of feature requests :)

As for the scenarios with different budgets you can always do a grid of optimisations with different budgets with current functionality. The optimisation itself does not make sense to have a flexible budget as it will always go for the maximum as we are optimising for KPI and media impact is expected to be always positive.

**Reply by pabloduque0** [reply]
FYI @yukaabe

**Reply by xijianlim** [reply]
Awesome, thanks for responding so promptly! I'm actually pretty excited about these developments. Great job guys.
